lucille-plugins/lucille-api/src/main/java/com/kmwllc/lucille/APIApplication.java

```diff
@@ -36,31 +39,60 @@ public String getName() {
   protected void bootstrapLogging() {}
 
   @Override
-  public void initialize(Bootstrap<LucilleAPIConfiguration> bootstrap) {}
+  public void initialize(Bootstrap<LucilleAPIConfiguration> bootstrap) {
+
+    bootstrap.addBundle(new SwaggerBundle<LucilleAPIConfiguration>() {
+      @Override
+      protected SwaggerBundleConfiguration getSwaggerBundleConfiguration(
+          LucilleAPIConfiguration configuration) {
+        return configuration.swaggerBundleConfiguration;
+      }
+    });
+
+  }
 
   @Override
   public void run(LucilleAPIConfiguration config, Environment env) throws Exception {
+    System.out.println(String.format("starting lucille-api from %s config %s env %s",
+        System.getProperty("user.dir"), config, env));
+
     RunnerManager runnerManager = RunnerManager.getInstance();
 
-    // Enable Basic Auth
-    if (config.getAuthConfig().getType().equals(AuthType.BASIC_AUTH)) {
-      env.jersey().register(new AuthDynamicFeature(
-          new BasicCredentialAuthFilter.Builder<PrincipalImpl>()
-              .setAuthenticator(new BasicAuthenticator(config.getAuthConfig().getPassword()))
-              .buildAuthFilter()
-      ));
+    boolean authEnabled = config.getAuthConfig().isEnabled();
+
+    // Enable Basic Auth only if it's enabled in the configuration
+    if (authEnabled) {
+      if (config.getAuthConfig().getType().equals(AuthType.BASIC_AUTH)) {
+        env.jersey()
+            .register(new AuthDynamicFeature(new BasicCredentialAuthFilter.Builder<PrincipalImpl>()
+                .setAuthenticator(new BasicAuthenticator(config.getAuthConfig().getPassword()))
+                .buildAuthFilter()));
+        env.jersey().register(new AuthValueFactoryProvider.Binder<>(PrincipalImpl.class));
+        log.info("Basic authentication has been enabled.");
+      } else {
+        throw new Exception("Unsupported auth type configured for the Lucille Admin API.");
+      }
     } else {
-      throw new Exception("No Auth configured for the Lucille Admin API.");
+      log.info("Authentication is disabled.");
     }
 
     // Register our 3 Resources
-    env.jersey().register(new LucilleResource(runnerManager));
+    AuthHandler authHandler = new AuthHandler(authEnabled);
+    env.jersey().register(new LucilleResource(runnerManager, authHandler));
     env.jersey().register(new LivenessResource());
     env.jersey().register(new ReadinessResource());
     env.jersey().register(new AuthValueFactoryProvider.Binder<>(PrincipalImpl.class));
   }
 
   public static void main(String[] args) throws Exception {
+    System.out.println(String.format("starting lucille-api from %s args %s",
```

Comment:
use the logger please
this is a codebase that avoids System.out.println and uses logging

Reply:
Heh, all other System.out statements in other files (which there were plenty of) were mistakes, but this one wasn't .. for the reason I gave @kwatters .. which was at the very beginning of an app startup, the startup details logged can change logging to hide reasons why its not logging.  But I understand I didn't convince anyone.. I'll update.

Reply:
fixed